### ISSUE TITLE
Watches fail on ThirdPartyResourceData objects

### ISSUE THREAD

**Issue by polvi** [kubernetes/kubernetes#24963]
```
$ kubectl get --watch foos
NAME          LABELS    DATA
polvi-foo   <none>    {"apiVersion":"coreos.com/v1","kind":"Foo...
$
```

The watch does not occur. Then the API server reports:

```
E0428 22:39:18.481109   92876 watch.go:209] unable to encode watch object: unexpected object to encode: &versioned.InternalEvent{Type:"ADDED", Object:(*runtime.Unknown)(0xc821438780)} (&streaming.encoder{writer:(*metrics.fancyResponseWriterDelegator)(0xc820168f08), encoder:(*thirdpartyresourcedata.thirdPartyResourceDataCodec)(0xc821208b00), buf:(*bytes.Buffer)(0xc820a8b3b0)})
```

To reproduce:
1. Create a ThirdPartyResource
2. Create an object of your new Kind
3. kubectl get foos --watch

**Comment by AdoHe** [comment]
I will try to reproduce this and take a look.

**Comment by AdoHe** [comment]
Seems this can reproduced on HEAD, will take a look.

**Comment by AdoHe** [comment]
@brendandburns seems thirdparty resource data codec can't handle `versioned.InternalEvent` [https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/thirdpartyresourcedata/codec.go#L391](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/thirdpartyresourcedata/codec.go#L391)

**Comment by brendandburns** [comment]
I'll work on this.

**Comment by nowprovision** [comment]
@brendandburns 

Ive created a quick fix for this to enable me to continue to work on Third pary resource clients
https://github.com/sitepod/kubernetes/commit/d0e9c029cbe7379f386fc2a6696824df604dfd0d

YMMV

**Comment by brendandburns** [comment]
@nowprovision can you send that as a PR?

**Comment by nowprovision** [comment]
@brendandburns Done. https://github.com/kubernetes/kubernetes/pull/25938

**Comment by polvi** [comment]
looks like this was fixed in #25938, so I'm closing.
